packages/app/src/systems/Settings/machines/settingsMachine.ts

```diff
@@ -0,0 +1,150 @@
+import type { InterpreterFrom, StateFrom } from 'xstate';
+import { assign, createMachine } from 'xstate';
+
+import type { AccountInputs } from '~/systems/Account';
+import { AccountService } from '~/systems/Account';
+import type { ChildrenMachine } from '~/systems/Core';
+import { FetchMachine } from '~/systems/Core';
+import type { UnlockMachine } from '~/systems/DApp';
+
+type MachineServices = {
+  getMnemonicPhrase: {
+    data: string[];
+  };
+};
+
+type MachineContext = {
+  words: string[];
+};
+
+type MachineEvents = (
+  | { type: 'UNLOCK_WALLET'; input: AccountInputs['unlock'] }
+  | { type: 'CHANGE_PASSWORD'; input: AccountInputs['changePassword'] }
+) & { data?: string[] };
+
+export const settingsMachine = createMachine(
+  {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    tsTypes: {} as import('./settingsMachine.typegen').Typegen0,
+    schema: {
+      context: {} as MachineContext,
+      services: {} as MachineServices,
+      events: {} as MachineEvents,
+    },
+    predictableActionArguments: true,
+    id: '(machine)',
+    initial: 'unlocking',
+    states: {
+      changingPassword: {
+        invoke: {
+          src: 'changePassword',
+          data: (_: MachineContext, ev: MachineEvents) => {
+            return { input: ev.input };
+          },
+          onDone: [
+            {
+              target: 'unlocking',
+              cond: FetchMachine.hasError,
+            },
+            {
+              target: 'passwordChanged',
+            },
+          ],
+        },
+      },
+      unlocking: {
+        tags: ['unlocking'],
+        on: {
+          UNLOCK_WALLET: {
+            target: 'gettingMnemonic',
+          },
+        },
+      },
+      gettingMnemonic: {
+        tags: ['unlocking'],
+        invoke: {
+          src: 'unlockAndGetMnemonic',
+          data: {
+            input: (_: MachineContext, ev: MachineEvents) => {
+              return ev.input;
+            },
```

Comment:
can we do it one line? 


```suggestion
            input: (_: MachineContext, ev: MachineEvents) => ev.input
```